frontend/src/app/services/auth/auth.service.ts

```diff
@@ -0,0 +1,65 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { JwtHelperService } from '@auth0/angular-jwt';
+import { BehaviorSubject, Observable, tap } from 'rxjs';
+import { DecodedToken } from 'src/app/interfaces/decoded-token';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  public api: string = environment.apiUrl;
+  private token: string = null;
+  private jwtHelper = new JwtHelperService();
```

Comment:
Angular in core uses Dependency Injection mechanism. You need to delegate creating of instances to Dependency Injection. You don't need to create instances in Angular manually. You should inject services using constructor